### PR TITLE
[TEVA-1036] Add Schools overview component

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1036

Screenshot 2020-09-04 at 10 03 35

### CODE REVIEW

File: app/components/jobseekers/school_group_overview_component.rb [hunk 1]
```diff
@@ -7,6 +7,6 @@ def initialize(vacancy:)
   end
 
   def render?
-    @vacancy.job_location == 'central_office'
+    @vacancy.central_office?
```
Comment (cpjmcquillan) [inline app/components/jobseekers/school_group_overview_component.rb:10]: 👍

File: spec/components/jobseekers/schools_overview_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+RSpec.describe Jobseekers::SchoolsOverviewComponent, type: :component do
+  let(:school_group) { create(:school_group) }
+  let(:school_1) { create(:school, name: 'Oxford University') }
+  let(:school_2) { create(:school, name: 'Cambridge University') }
+  let(:school_3) { create(:school, name: 'London School of Economics') }
+  let(:vacancy) { create(:vacancy, :at_multiple_schools) }
+  let(:vacancy_presenter) { VacancyPresenter.new(vacancy) }
+
+  before do
+    school_group.school_group_memberships.create(school: school_1)
```
Comment (cpjmcquillan) [inline spec/components/jobseekers/schools_overview_component_spec.rb:12]: This is much better, nice!

File: app/components/jobseekers/schools_overview_component.html.haml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        %tr.govuk-table__row
+          %td.govuk-table__cell{ class: "govuk-!-font-weight-bold" }
+            = t('jobs.contact_email')
+          %td.govuk-table__cell
+            = mail_to @vacancy.contact_email, @vacancy.contact_email, class: 'govuk-link link-wrap', subject: t('jobs.contact_email_subject', job: @vacancy.job_title), body: t('jobs.contact_email_body', url: job_url(@vacancy))
+
+  - if @vacancy.about_school.present?
+    %h4.govuk-heading-s
+      = t('schools.about_schools')
+    %p= vacancy_or_organisation_description(@vacancy)
```
Comment (cpjmcquillan) [inline app/components/jobseekers/schools_overview_component.html.haml:22]: This should just be `@vacancy.about_school` I think

File: app/components/jobseekers/schools_overview_component.html.haml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+  %h4.govuk-heading-s
+    = t('schools.school_details')
+
+  #accordion-default.govuk-accordion{"data-module" => "govuk-accordion"}
```
Comment (cpjmcquillan) [inline app/components/jobseekers/schools_overview_component.html.haml:27]: `data: { module: "govuk-accordion" }` if you prefer

File: app/components/jobseekers/schools_overview_component.html.haml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    - @vacancy.organisations.each_with_index do |organisation, index|
+      .govuk-accordion__section
+        .govuk-accordion__section-header
+          %h3.govuk-accordion__section-heading
+            %span.govuk-accordion__section-button{ id: "accordion-default-heading-#{index}"}
+              = organisation.name
+        .govuk-accordion__section-content{ id: "accordion-default-content-#{index}", aria: { labelledby: "accordion-default-heading-1" } }
```
Comment (alexbowen) [inline app/components/jobseekers/schools_overview_component.html.haml:34]: i think this should be `"accordion-default-heading-#{index}` so it matches the id above

File: spec/components/jobseekers/schools_overview_component_spec.rb [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    school_group.school_group_memberships.create(school: school_2)
+    school_group.school_group_memberships.create(school: school_3)
+    vacancy.organisation_vacancies.create(organisation: school_1)
+    vacancy.organisation_vacancies.create(organisation: school_2)
+    vacancy.organisation_vacancies.create(organisation: school_3)
+    render_inline(described_class.new(vacancy: vacancy_presenter))
+  end
+
+  describe '#render?' do
+    context 'when vacancy is at a trust head office' do
```
Comment (cpjmcquillan) [inline spec/components/jobseekers/schools_overview_component_spec.rb:22]: `when vacancy job_location is central_office`

**Review by cpjmcquillan — APPROVED** [review]

File: app/components/jobseekers/schools_overview_component.html.haml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+              %tr.govuk-table__row
+                %td.govuk-table__cell{ class: "govuk-!-font-weight-bold" }
+                  = t('schools.website')
+                %td.govuk-table__cell
+                  = link_to("#{organisation.name} website (opens in a new window)", organisation.url, class: 'govuk-link link-wrap', target: '_blank')
+
+  - if @vacancy.school_visits.present?
+    %h4.govuk-heading-s
+      = t('schools.schools_visits')
+    %p= @vacancy.school_visits
```
Comment (cpjmcquillan) [inline app/components/jobseekers/schools_overview_component.html.haml:80]: I think the presenter adds the <p> element already

**Review by alexbowen — APPROVED** [review]

**Pull request merged**